pyro/infer/trace_elbo.py

```diff
@@ -112,7 +112,7 @@ def loss_and_grads(self, model, guide, *args, **kwargs):
                         guide_site = guide_trace.nodes[name]
                         guide_log_prob, score_function_term, entropy_term = guide_site["score_parts"]
 
-                        elbo_particle = elbo_particle + model_log_prob_sum - guide_log_prob.sum()
+                        elbo_particle = elbo_particle + (model_log_prob_sum.item() - guide_log_prob.sum().item())
```

Comment:
why was this needed?

Reply:
This fixes a tensor->float conversion that I missed back in #848. Everywhere else we assume that `SVI.step()` returns a Python float as loss. This fix also increases the precision of the accumulator from (typically) 32bits to 64bit floats. I've added it to this PR because I'm testing the loss values and realized they weren't `float`s as expected.